### ISSUE TITLE
Improve error reporting: Missing "else" branch for if expression

### ISSUE THREAD

**Issue by isaacabraham** [Microsoft/visualfsharp#1104]
## What
Developers coming from other statement based languages often fall foul of the following: -

```fsharp
let x = 10
let y =
   if x > 10 then "test"
```
As if is an expression in F#, you need to provide an "else" branch as well. The error message is not helpful: -

```
error FS0001: This expression was expected to have type
    unit    
but here has type
    string   
```

## Why
Developers coming from other languages are used to working with statements, not expressions, where it is legal to only have a single branch of code in a conditional. It's not obvious from the current error message what the issue is; in fact the solution suggests that you should ignore the result of the branch in order that it gives unit back!

## How
A better error message might be something like: -

``You have not supplied the "else" case for this if expression. If / then is an expression and so must always return some result in all cases"``

This should also apply to if / elif / elif with a missing "else".

**Comment by forki** [comment]
![mine](https://cloud.githubusercontent.com/assets/57396/14935177/b13b1fe4-0ec8-11e6-8062-25584bea93b5.gif)

**Comment by forki** [comment]
WIP PR in https://github.com/Microsoft/visualfsharp/pull/1142

**Comment by KevinRansom** [comment]
MissingElse Resolved with: #1142
